release/release.go

```diff
@@ -30,51 +30,75 @@ import (
 // Series holds the Ubuntu Core series for snapd to use.
 var Series = "16"
 
-// LSB contains the /etc/os-release information of the system.
-type LSB struct {
+// OSRelease contains information about the system extracted from /etc/os-release.
+type OSRelease struct {
 	ID       string
+	Name     string
 	Release  string
 	Codename string
 }
 
-var lsbReleasePath = "/etc/os-release"
+var osReleasePath = "/etc/os-release"
 
-// ReadLSB returns the os-release information of the current system.
-func ReadLSB() (*LSB, error) {
-	lsb := &LSB{}
+// ReadOSRelease returns the os-release information of the current system.
+func ReadOSRelease() (*OSRelease, error) {
+	osRelease := &OSRelease{}
 
-	content, err := ioutil.ReadFile(lsbReleasePath)
+	content, err := ioutil.ReadFile(osReleasePath)
 	if err != nil {
 		return nil, fmt.Errorf("cannot read os-release: %s", err)
 	}
 
 	for _, line := range strings.Split(string(content), "\n") {
+		if strings.HasPrefix(line, "ID=") {
+			tmp := strings.SplitN(line, "=", 2)
+			osRelease.ID = strings.Trim(tmp[1], "\"")
+		}
 		if strings.HasPrefix(line, "NAME=") {
 			tmp := strings.SplitN(line, "=", 2)
-			lsb.ID = strings.Trim(tmp[1], "\"")
+			osRelease.Name = strings.Trim(tmp[1], "\"")
 		}
 		if strings.HasPrefix(line, "VERSION_ID=") {
 			tmp := strings.SplitN(line, "=", 2)
-			lsb.Release = strings.Trim(tmp[1], "\"")
+			osRelease.Release = strings.Trim(tmp[1], "\"")
 		}
 		if strings.HasPrefix(line, "UBUNTU_CODENAME=") {
 			tmp := strings.SplitN(line, "=", 2)
-			lsb.Codename = strings.Trim(tmp[1], "\"")
+			osRelease.Codename = strings.Trim(tmp[1], "\"")
 		}
 	}
-	if lsb.Codename == "" {
-		lsb.Codename = "xenial"
+	if osRelease.Codename == "" {
+		osRelease.Codename = "xenial"
 	}
 
-	return lsb, nil
+	return osRelease, nil
 }
 
 // OnClassic states whether the process is running inside a
 // classic Ubuntu system or a native Ubuntu Core image.
 var OnClassic bool
 
+// ReleaseInfo contains data loaded from /etc/os-release on startup.
+var ReleaseInfo OSRelease
```

Comment:
It's strange to have both ReadOSRelease and OSRelease here. Seems fine to have it as a variable, as long as we can make the reading function private.